datasets/s2orc/s2orc.py

```diff
@@ -0,0 +1,145 @@
+                "s2Url": datasets.Value("string"),
+                "pdfUrls": datasets.Sequence(datasets.Value("string")),
+                "s2PdfUrl": datasets.Value("string"),
+                "authors": [
+                    {
+                        "name": datasets.Value("string"),
+                        "ids": datasets.Sequence(datasets.Value("string")),
+                    },
+                ],
+                "inCitations": datasets.Sequence(datasets.Value("string")),
+                "outCitations": datasets.Sequence(datasets.Value("string")),
+                "fieldsOfStudy": datasets.Sequence(datasets.Value("string")),
+                "year": datasets.Value("int32"),
+                "venue": datasets.Value("string"),
+                "journalName": datasets.Value("string"),
+                "journalVolume": datasets.Value("string"),
+                "journalPages": datasets.Value("string"),
+                "sources": datasets.Sequence(datasets.Value("string")),
+                "doi": datasets.Value("string"),
+                "doiUrl": datasets.Value("string"),
+                "pmid": datasets.Value("string"),
+                "magId": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        _MANIFEST_URL = _ROOT_URL + "manifest.txt"
+        manifest_file = dl_manager.download_and_extract(_MANIFEST_URL)
+
+        file = open(manifest_file, "r", encoding="utf-8")
+        train_names = file.read().splitlines()
+
+        r = re.compile("(?s:s2\\-corpus\\-.*\\.gz)\\Z")  # files are of the form 's2-corpus-*.gz'
+        train_names = list(filter(r.match, train_names))
+
+        train_names = train_names[:101]  # comment this line to run on all 6000 files
+
+        train_names = [dl_manager.download_and_extract(_ROOT_URL + x) for x in train_names]
+
+        with open(manifest_file, "wb") as wfd:  # added all the data to already downloaded manifest file
+            for f in train_names:
+                with open(f, "rb") as fd:
+                    shutil.copyfileobj(fd, wfd)
```

Comment:
Instead of writing the content of all the files you can simply iter through each file in `_generate_examples` instead.
To get the paths of all the files in `_generate_examples` you can pass them to `gen_kwargs`.

For example
```python
gen_kwargs={
    "filepaths": train_filepaths,
}
```